cmd/ui/src/rendering/programs/edge.curvedArrowHead.ts

```diff
@@ -150,9 +151,16 @@ export default class CurvedEdgeArrowHeadProgram extends AbstractEdgeProgram {
             return;
         }
 
+        let start, end: Coordinates;
+        if (data.direction === EdgeDirection.BACKWARDS) {
+            start = { x: targetData.x, y: targetData.y };
+            end = { x: sourceData.x, y: sourceData.y };
+        } else {
+            start = { x: sourceData.x, y: sourceData.y };
+            end = { x: targetData.x, y: targetData.y };
+        }
```

Comment:
I'm curious why we have the notion of an edge direction? Isn't the direction of an edge simply based on its source and target nodes?

Reply:
That is a good question and a good point. It could possibly be clearer but this is related to how the edges are first initially grouped for applying a different offset when there are multiple edges between two nodes as they are gathered together in a key value pair regardless of which the start or the end node are. It avoids creating overlapping edges. Here is where that happens: https://github.com/SpecterOps/BloodHound/blob/main/cmd/ui/src/views/Explore/utils.ts#L159

Reply:
I see, regardless of the direction of an edge we want to be able to group them nicely. Makes sense!